chunkenc/xor.go

```diff
@@ -183,8 +183,11 @@ func (a *xorAppender) Append(t int64, v float64) {
 	a.tDelta = tDelta
 }
 
-func bitRange(x int64, nbits uint8) bool {
-	return -((1<<(nbits-1))-1) <= x && x <= 1<<(nbits-1)
+func bitRange(x int64, nbits int) bool {
+	if x < 0 {
+		x = (-x) + 1
+	}
+	return bits.Len64(uint64(x-1)) < nbits
```

Comment:
Did you run a benchmark for comparison here? (Do we have one actually adding some variance to timestamps?)

Adding a branch in this code path is quite risky. But might well be it can be predicated sufficiently most of the time.

Reply:
I ran the tsdb command and saw no significant difference. I'll look at benchmarking in more detail.